chia/data_layer/data_store.py

```diff
@@ -821,3 +879,223 @@ async def get_operations(
+                                    right_hash = hash.hex()
+                                    while right_hash in add_to_db_cache:
+                                        node, left_hash = add_to_db_cache[right_hash]
+                                        del add_to_db_cache[right_hash]
+                                        await self._insert_node(
+                                            node,
+                                            NodeType.INTERNAL,
+                                            left_hash,
+                                            right_hash,
+                                            None,
+                                            None,
+                                            tree_id.hex(),
+                                        )
+                                        internal_nodes += 1
+                                        right_hash = node
+                                else:
+                                    raise RuntimeError(
+                                        f"Did not received expected node. Expected: {node} Received: {hash.hex()}"
+                                    )
+                                if len(stack) > 0:
+                                    node = stack.pop()
+                                else:
+                                    node = None
+                            else:
+                                left_hash = row["left"]
+                                right_hash = row["right"]
+                                left_hash_bytes = hexstr_to_bytes(left_hash)
+                                right_hash_bytes = hexstr_to_bytes(right_hash)
+                                hash = Program.to((left_hash_bytes, right_hash_bytes)).get_tree_hash(
+                                    left_hash_bytes, right_hash_bytes
+                                )
+                                if hash.hex() == node:
+                                    add_to_db_cache[right_hash] = (node, left_hash)
+                                    # At most max_height nodes will be pending to be added to DB.
+                                    assert len(add_to_db_cache) <= 100
+                                else:
+                                    raise RuntimeError(
+                                        f"Did not received expected node. Expected: {node} Received: {hash.hex()}"
+                                    )
+                                stack.append(right_hash)
+                                node = left_hash
+
+                        if node is not None:
+                            raise RuntimeError("Did not download full data.")
+                        await ws.send_str(json.dumps({"type": "close"}))
+
+                await self._insert_root(
+                    bytes32.from_hexstr(root_json["tree_id"]),
+                    bytes32.from_hexstr(root_json["node_hash"]),
+                    Status(root_json["status"]),
```

Comment:
we would only download changes that are confirmed on chain so status here should always be CONFIRMED